packages/core/src/build/configAndIndexingFunctions.ts

```diff
@@ -73,6 +74,36 @@ export async function buildConfigAndIndexingFunctions({
 }> {
   const logs: { level: "warn" | "info" | "debug"; msg: string }[] = [];
 
+  const perNetworkLatestBlockNumber = new Map<string, Promise<number>>();
+
+  const resolveBlockNumber = async (
+    blockNumberOrTag: number | "latest" | undefined,
+    network: Network,
+  ) => {
+    if (blockNumberOrTag === undefined) {
+      return undefined;
+    }
+
+    if (Number.isNaN(blockNumberOrTag)) {
+      return undefined;
+    }
+    if (blockNumberOrTag === "latest") {
+      if (perNetworkLatestBlockNumber.has(network.name)) {
+        return perNetworkLatestBlockNumber.get(network.name)!;
+      } else {
+        const blockPromise = network.transport
```

Comment:
We should be thoughtful about error handling here. This is the first risky async work we're doing in this function. Worth a mocked test IMO.